Week1/homework/index.js

```diff
@@ -0,0 +1,205 @@
+const { promisify } = require('util');
+
+const app = express();
+app.use(bodyParser.json());
+
+const data = require('./cities.json');
+
+const connection = mysql.createConnection({
+  host: 'localhost',
+  user: 'root',
+  password: 'password',
+  database: 'world',
+});
+
+connection.connect();
+
+const createCountries = `CREATE TABLE countries (
+  Name varchar(254) ,
+  Continent varchar(50),
+  Region varchar(50) ,
+  SurfaceArea float(2) ,
+  IndepYear varchar(50) ,
+  Population float(2)  ,
+  LifeExpectancy varchar(50) ,
+  GNP float(2) ,
+  GNPOld varchar(50) ,
+  LocalName varchar(50),
+  GovernmentForm varchar(50),
+  HeadOfState varchar(50) ,
+  Capital varchar(50)
+);`;
+
+const createCities = `CREATE TABLE cities(
+  ID int NOT NULL AUTO_INCREMENT,
+  Name varchar(50),
+  CountryCode varchar(50),
+  District varchar (50),
+  Population int,
+  PRIMARY KEY (ID)
+);`;
+
+app.get('/createDatabase', (req, res) => {
+  connection.query('CREATE DATABASE world;', (error, results, fields) => {
+    if (error) {
+      console.log(error);
+      res.send(`Database hasn't created ${error}`);
+      connection.end();
+    }
+    console.log(`Database created`);
+    res.send(`Database created`);
+  });
+});
+
+app.put('/createCountries', (req, res) => {
+  connection.query(createCountries, (error, results, fields) => {
+    if (error) {
+      console.log(error);
+      res.send(`Table hasn't created ${error}`);
+      connection.end();
+    }
+    console.log(`Table created`);
+    res.send(`Table created`);
+  });
+});
+
+app.put('/createCities', (req, res) => {
+  connection.query(createCities, (error, results, fields) => {
+    if (error) {
+      console.log(error);
+      res.send(`Table hasn't created ${error}`);
+      connection.end();
+    }
+    console.log(`Table created`);
+    res.send(`Table created`);
+  });
+});
+
+app.post('/appendCountries', async (req, res) => {
+  try {
+    const { countries } = data;
+    for (const country of countries) {
+      const try1 = `INSERT INTO countries VALUES('${country.name}', '${country.continent}', '${country.region}', '${country.surface_area}', ' ${country.indep_year}', ' ${country.population}', ' ${country.life_expectancy}', ' ${country.gnp}', ' ${country.gnp_old}', ' ${country.local_name}', ' ${country.government_form}', ' ${country.head_of_state}', ' ${country.capital}');`;
+      await connection.query(try1, (error, results, fields) => {
+        if (error) {
+          console.log(`Table hasn't values ${error}`);
+        }
+      });
+    }
+    console.log('Table has values ');
+    res.write(`Success `);
+  } catch (err) {
+    res.write(`Error: ${err}`);
+  }
+  res.write(`Table has values`);
+  res.end();
+});
+
+app.post('/appendCities', async (req, res) => {
+  try {
+    const { cities } = data;
+    for (const city of cities) {
+      const try1 = `INSERT INTO cities values(NULL, '${city.name}', '${city.country_code}', '${city.district}', ' ${city.population}'); `;
```

Comment:
Good for the use of auto-increment.
After next week you will have to use prepared query to avoid attack on your code :)

Reply:
okay I will :) Thanks 👍